include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -150,14 +134,10 @@ auto _search_trivial(auto it, auto const & query, uint64_t const query_pos, sear
                     // Match
                     if (error_left.substitution == 0 && it.last_char() == query[query_pos])
                     {
-                        if constexpr (abort_on_hit)
-                        {
-                            if (_search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate))
-                                return true;
-                        }
-                        else
+                        if (_search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate) &&
+                            abort_on_hit)
```

Comment:
Is abort_on_hit only every evaluated at run-time now? If yes, it should be a regular parameter.

Reply:
I might change that when I replace this file with optimum search schemes. :+1: